Form/FormFlow.php

```diff
@@ -609,6 +645,16 @@ protected function bindFlow() {
 			$reset = true;
 		}
 
+		if ($this->getRequest()->isMethod('POST') && ! $this->dataManager->hasInstance($this)) {
+			// Form is stale, kill posted data and reset
+			$this->getRequest()->request->replace();
+			$reset = true;
+			$this->setStale(true);
+
+                        // Regenerate instance ID
+			$this->setInstanceId($this->determineInstanceId());
```

Comment:
When the code is meant to target such "tricky case", then there should be a test for it to avoid future regressions.